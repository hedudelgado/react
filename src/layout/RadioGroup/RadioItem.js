import React from 'react';
import {
  bool,
  string,
  array,
  object,
  oneOfType,
  number,
} from 'prop-types';
import classnames from 'classnames';

const RadioItem = ({
  label,
  subLabels,
  field: {
    name,
    onBlur,
    onChange,
    value: fieldValue,
  },
  value,
  id,
  sessioncamhide,
  ...props
}) => {
  const labels = subLabels.map(subLabel => <span key={subLabel} className="wb-radio__label__inner-info">{subLabel}</span>);
  const classes = classnames({
    sessioncamhidetext: sessioncamhide,
  });
  return (
    <li className="wb-radio__inner">
      <label htmlFor={id} className="wb-radio__label">
        <input
          type="radio"
          name={name}
          id={id}
          onBlur={onBlur}
          onChange={onChange}
          checked={value === fieldValue}
          value={value}
          className={classes}
          {...props}
        />
        <span className="wb-radio__label__inner">
          <span className="wb-radio__label__inner__title">{label}</span>
          {labels}
        </span>
      </label>
    </li>
  );
};

RadioItem.propTypes = {
  value: oneOfType([
    string,
    number,
  ]),
  label: oneOfType([
    string,
    number,
  ]),
  subLabels: array,
  field: object,
  id: oneOfType([
    string,
    number,
  ]),
  /**
    Sessioncam field masking.
  */
  sessioncamhide: bool,
};

RadioItem.defaultProps = {
  subLabels: [],
};

export default RadioItem;
