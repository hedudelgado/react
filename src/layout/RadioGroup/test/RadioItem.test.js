import React from 'react';
import { shallow } from 'enzyme';

import RadioItem from '../RadioItem';

describe('RadioItem Component', () => {
  const props = {
    value: 'true',
    label: 'This is my test label',
    field: {
      name: 'test',
    },
    id: 'test',
    subLabels: [
      'test sub label 1',
      'test sub label 2',
    ],
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RadioItem {...props} />);
  });

  describe('Renders RadioItem Element', () => {
    it('should render the component', () => {
      expect(wrapper).not.toBeEmptyRender();
      expect(wrapper).toMatchSnapshot();
    });

    it('should have input type radio', () => {
      expect(wrapper).toContainExactlyOneMatchingElement('input[type="radio"]');
    });

    it('should have span label with correct content', () => {
      expect(wrapper).toContainExactlyOneMatchingElement('.wb-radio__label__inner__title');
      expect(wrapper.find('.wb-radio__label__inner__title')).toHaveText('This is my test label');
    });

    it('should have htmlFor property with correct name', () => {
      expect(wrapper).toContainExactlyOneMatchingElement('[htmlFor="test"]');
    });

    it('should have two subLabels', () => {
      expect(wrapper).toContainMatchingElements(2, '.wb-radio__label__inner-info');
    });
  });
});
