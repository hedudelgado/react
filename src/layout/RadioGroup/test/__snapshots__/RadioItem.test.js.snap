// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioItem Component Renders RadioItem Element should render the component 1`] = `
<li
  className="wb-radio__inner"
>
  <label
    className="wb-radio__label"
    htmlFor="test"
  >
    <input
      checked={false}
      className=""
      id="test"
      name="test"
      type="radio"
      value="true"
    />
    <span
      className="wb-radio__label__inner"
    >
      <span
        className="wb-radio__label__inner__title"
      >
        This is my test label
      </span>
      <span
        className="wb-radio__label__inner-info"
        key="test sub label 1"
      >
        test sub label 1
      </span>
      <span
        className="wb-radio__label__inner-info"
        key="test sub label 2"
      >
        test sub label 2
      </span>
    </span>
  </label>
</li>
`;
